### ISSUE TITLE
How use jtable in php mvc1 (zend framework)?

### ISSUE THREAD

**Issue by del34** [hikalkan/jtable#1828]
hi 
when i use jtable in simple php projects its work fine but when i change it to php mvc  i got "An error occured while communicating to the server." and it didnt show any data  in jtable.

 but in the chrom console it print json:

{"Result":"OK","TotalRecordCount":"20","Records":[{"PersonId":"1","0":"1","Name":"Benjamin Button","1":"Benjamin Button","Age":"17","2":"17","RecordDate":"2011-12-27 00:00:00","3":"2011-12-27 00:00:00"},{"PersonId":"12","0":"12","Name":"damir","1":"damir","Age":"27","2":"27","RecordDate":"2015-09-17 12:17:53","3":"2015-09-17 12:17:53"}]}

that shows every thing is correct. but jtable didnt show anything. i really dont know what should i do. please help i'm so confused and dont know what should i do. shoud i change any thing in js files?

in  adminlayout.phtml:
 i use other jquery function in $(document).ready.but when i delete them it didnt work too.

$(document).ready(function () {
    //Prepare jTable
	$('#PeopleTableContainer').jtable({
		title: 'Table of people',
		paging: true,
		pageSize: 2,
		sorting: true,
		defaultSorting: 'Name',
		actions: {
			listAction: root_url+'admin/index/personpagedsorted?action=list',
			},
		fields: {
			PersonId: {
				key: true,
				create: false,
				edit: false,
				list: false
			},
			Name: {
				title: 'Author Name',
				width: '40%'
			},
			Age: {
				title: 'Age',
				width: '20%'
			},
			RecordDate: {
				title: 'Record date',
				width: '30%',
				type: 'date',
				create: false,
				edit: false
			}
			
		}

		 
	});

	//Load person list from server
	
	$('#PeopleTableContainer').jtable('load');
    
in controller: i only use list action.

public function personpagedsortedAction()
     {
        
         if ($_GET["action"]=="list")
        {
            $dbh=Zend_Registry::get('db');
            $sth = $dbh->prepare("SELECT COUNT(*) AS RecordCount FROM people;");
            $sth->execute();
            $row = $sth->fetch(PDO::FETCH_BOTH);
             
             $recordCount = $row['RecordCount'];
            $jtSorting=$_GET["jtSorting"];
           $jtStartIndex= $_GET["jtStartIndex"];
          $jtPageSize= $_GET["jtPageSize"];
            //Get records from database
           
         //  $result =$db->personselectinput($jtSorting, $jtStartIndex, $jtPageSize);
            
           $str_sql="SELECT * FROM people ORDER BY " . $jtSorting . " LIMIT " . $jtStartIndex . "," . $jtPageSize . ";";
           // $res=$db->query($str_sql);
           // $res=$db->fetchAll($str_sql);
            
           $sth = $dbh->prepare($str_sql);
           $sth->execute();
           
            //Add all records to an array
            $rows = array();
        
           while ( $result = $sth->fetch(PDO::FETCH_BOTH))
           {
               $rows[]=$result;
           }
           
            //Return result to jTable
            $jTableResult = array();
            $jTableResult['Result'] = "OK";
            $jTableResult['TotalRecordCount'] = $recordCount;
            $jTableResult['Records'] = $rows;
         // i change print to echo and return but it didnt work too

          print json_encode($jTableResult);

        }
   }  in edittest.phtml:i wrote div id="PeopleTableContainer"> that show the table without any data

**Comment by del34** [comment]
in edittest.phtml: 
<div id="PeopleTableContainer"> </div>

**Comment by apb1963** [comment]
Some general troubleshooting tips:
First, backup your code.  I can't stress this enough.  Simplify & isolate the problem.  Remove any code not directly related to the problem.  When you have reduced the code to the absolute minimum needed to reproduce the problem, then post that here and maybe someone else will take the time to analyze it (I'm a beginner so other then what I've stated here, I can't help).  You will probably find you're able to figure it out yourself once you've followed these tips.

Also, I've learned that jtable is apparently referring to the database server when it says "server".  Check all your data for mismatched and/or missing fields and anything else data related that might be wrong, especially punctuation (braces, commas, semicolons, etc.).   Print the SQL statement after it gets all filled in and make sure it says what you think it should say - then run your SQL statement in mysql directly and verify it works and returns what you expect.  Obviously you want to test these things with mvc since you state that's where your problem is.
